packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
+      });
+    }, /Specifying `defaultLayout` to .+ is deprecated\./);
+  }
 
-QUnit.test('Assigning defaultLayout to a component should set it up as a layout if no layout was found [DEPRECATED]', function() {
-  expect(2);
+  ['@test Assigning defaultLayout to a component should set it up as a layout if layout was found [DEPRECATED]'](assert) {
+    assert.expect(2);
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
 
-  expectDeprecation(() => {
-    boot(() => {
-      appInstance.register('controller:application', Controller.extend({
-        'text': 'outer'
-      }));
-
-      appInstance.register('component:my-component', Component.extend({
-        text: 'inner',
-        defaultLayout: compile('{{text}}-{{yield}}')
-      }));
-    });
-  }, /Specifying `defaultLayout` to .+ is deprecated\./);
-
-  equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
-});
-
-QUnit.test('Assigning defaultLayout to a component should set it up as a layout if layout was found [DEPRECATED]', function() {
-  expect(2);
-
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
-  setTemplate('components/my-component', compile('{{text}}-{{yield}}'));
-
-  expectDeprecation(() => {
-    boot(() => {
-      appInstance.register('controller:application', Controller.extend({
-        'text': 'outer'
-      }));
+    this.add('controller:application', Controller.extend({
+      text: 'outer'
+    }));
 
-      appInstance.register('component:my-component', Component.extend({
+    this.addComponent('my-component', {
+      ComponentClass: Component.extend({
         text: 'inner',
-        defaultLayout: compile('should not see this!')
-      }));
+        defaultLayout: this.compile('should not see this!')
+      }),
+      template: '{{text}}-{{yield}}'
     });
-  }, /Specifying `defaultLayout` to .+ is deprecated\./);
 
-  equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
-});
+    expectDeprecation(() => {
+      this.visit('/').then(() => {
+        let text = this.$('#wrapper').text().trim();
+        assert.equal(text, 'inner-outer', 'The component is composed correctly');
+      });
+    }, /Specifying `defaultLayout` to .+ is deprecated\./);
+  }
 
-QUnit.test('Using name of component that does not exist', function () {
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#no-good}} {{/no-good}}</div>'));
+  ['@test Using name of component that does not exist'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#no-good}} {{/no-good}}</div>');
 
-  expectAssertion(() => boot(), /.* named "no-good" .*/);
+    expectAssertion(() => this.visit('/'), /.* named "no-good" .*/);
```

Comment:
I expect this is falling prey to https://github.com/emberjs/ember.js/issues/15327 and failing. But if you write it as an autoboot test, it will work just fine. Since the rest of the file should probably be autoboot, making this as well seems good.

If you wanted to add a comment ala https://github.com/emberjs/ember.js/blob/1c720c8d4f5ec85036eff50289d7cd334c96ec37/packages/ember/tests/helpers/link_to_test.js#L1518-L1536 I think that would be good. At least mention the issue number.